csrc/layernorm_kernels.cu

```diff
@@ -60,13 +75,21 @@ __global__ void fused_add_rms_norm_kernel(
   __syncthreads();
 
   for (int idx = threadIdx.x; idx < hidden_size; idx += blockDim.x) {
-    float x = (float) residual[blockIdx.x * hidden_size + idx];
+    float x = use_shmem?shmem[idx]:residual[blockIdx.x * hidden_size + idx];
     input[blockIdx.x * hidden_size + idx] = ((scalar_t) (x * s_variance)) * weight[idx];
   }
 }
 
 } // namespace vllm
 
+
+inline int getMaxSharedMemoryPerBlock(const torch::Tensor& input) {
+  int max_shmem_size;
+  cudaDeviceGetAttribute(&max_shmem_size, cudaDevAttrMaxSharedMemoryPerBlock, input.device().index());
+  return max_shmem_size;
+}
```

Comment:
A quick question: is this compatible with AMD GPUs?

Reply:
working on vector types. not tested on AMD yet.  It seems not compatible with AMD. I will fix this as well